src/functional/iterator/transforms/global_tmps.py

```diff
@@ -250,7 +252,7 @@ def _named_range_with_offsets(
 
 
 def _extend_cartesian_domain(
-    domain: ir.FunCall, offsets: Sequence[tuple], offset_provider: Mapping[str, CartesianAxis]
+    domain: ir.Expr, offsets: Sequence[tuple], offset_provider: Mapping[str, CartesianAxis]
```

Comment:
Would it make sense to pass the proper thing on the call-side? Because on line 259, we assert that it's a FunCall.